pkg/vendir/directory/staging_dir.go

```diff
@@ -58,7 +61,74 @@ func (d StagingDir) NewChild(path string) (string, error) {
 	return childPath, nil
 }
 
+func (d StagingDir) CopyExistingFiles(rootDir string, stagingPath string, contents ctlconf.DirectoryContents) error {
+
+	if len(contents.IgnorePaths) == 0 {
+		return nil
+	}
+
+	// Create reference point from staging path to root
+	rootPath := strings.Replace(stagingPath, d.stagingDir, rootDir, 1)
+
+	if _, err := os.Stat(rootPath); os.IsNotExist(err) {
+		return nil // Path does not exist so there is nothing to copy
+	}
+
+	var ignorePaths []string
+	for _, ignorePath := range contents.IgnorePaths {
+		ignorePaths = append(ignorePaths, filepath.Join(rootPath, ignorePath)) // Prefix ignore glob with destination path
+	}
+
+	// Consider WalkDir in the future for efficiency (Go 1.16)
+	// Walk root path above to determine files that can be ignored
+	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Verify that the path should be ignored
+		if !ignorePath(path, ignorePaths) {
+			return nil
+		}
+
+		stagingPath := strings.Replace(path, rootPath, stagingPath, 1) // Preserve structure from destination to staging
+
+		// Ensure that the directories exist in the staging directory
+		stagingDir := filepath.Dir(stagingPath)
+		err = os.MkdirAll(stagingDir, 0700)
+		if err != nil {
+			return fmt.Errorf("Unable to create staging directory '%s': %s", stagingDir, err)
+		}
+
+		// Move the file to the staging directory
+		err = os.Rename(path, stagingPath)
```

Comment:
This was a good idea, I was able to create a `copy` function. For future proofing I did not use the `ioutil` package since it is now deprecated as of 1.16.